app/src/main/java/ch/epfl/sdp/blindly/localDB/UserDAO.kt

```diff
@@ -0,0 +1,204 @@
+    @Query("SELECT birthday FROM userentity WHERE uid = :id")
+    fun getUserBirthday(id: String): String?
+
+    /**
+     * Get the gender from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's gender
+     */
+    @Query("SELECT gender FROM userentity WHERE uid = :id")
+    fun getUserGender(id: String): String?
+
+    /**
+     * Get the sexualOrientation from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's sexualOrientation
+     */
+    @Query("SELECT sexualOrientations FROM userentity WHERE uid = :id")
+    fun getUserSexualOrientation(id: String): List<String>?
+
+    /**
+     * Get the showMe from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's showMe
+     */
+    @Query("SELECT showMe FROM userentity WHERE uid = :id")
+    fun getUserShowMe(id: String): String?
+
+    /**
+     * Get the passions from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's passions
+     */
+    @Query("SELECT passions FROM userentity WHERE uid = :id")
+    fun getUserPassions(id: String): List<String>?
+
+    /**
+     * Get the radius from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's radius
+     */
+    @Query("SELECT radius FROM userentity WHERE uid = :id")
+    fun getUserRadius(id: String): Int?
+
+    /**
+     * Get the matches from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's matches
+     */
+    @Query("SELECT matches FROM userentity WHERE uid = :id")
+    fun getUserMatches(id: String): List<String>?
+
+    /**
+     * Get the likes from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's likes
+     */
+    @Query("SELECT likes FROM userentity WHERE uid = :id")
+    fun getUserLikes(id: String): List<String>?
+
+    /**
+     * Get the description from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's description
+     */
+    @Query("SELECT description FROM userentity WHERE uid = :id")
+    fun getUserDescription(id: String): String?
+
+    /**
+     * Get the recordingPath from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's recordingPath
+     */
+    @Query("SELECT recordingPath FROM userentity WHERE uid = :id")
+    fun getUserRecordingPath(id: String): String?
+
+    /**
+     * Get the ageRange from a given uid
+     *
+     * @param id the uid to find the user
+     * @return the user's ageRange
+     */
+    @Query("SELECT agerange FROM userentity WHERE uid = :id")
+    fun getUserAgeRangeValue(id: String): String?
+
+    fun getUserAgeRange(id: String): List<Int> {
+        val ageRange = getUserAgeRangeValue(id)?.split(",")
+        if(ageRange != null) {
+            return listOf(ageRange[0].toInt(), ageRange[1].toInt())
+        }
+        return listOf(18, 99)
```

Comment:
Here as well, MIN_AGE and MAX_AGE would be nicer.

Reply:
done !